pkg/apis/monitoring/v1/types.go

```diff
@@ -48,6 +48,14 @@ type ByteSize string
 // +kubebuilder:validation:Pattern:="^(0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?)$"
 type Duration string
 
+// ImprovedDuration is a valid time duration that can be parsed by Prometheus model.ParseDuration() function.
+// Compared to Duration, ImprovedDuration enforces a minimum length of 1.
+// Supported units: y, w, d, h, m, s, ms
+// Examples: `30s`, `1m`, `1h20m15s`, `15d`
+// +kubebuilder:validation:Pattern:="^(0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?)$"
+// +kubebuilder:validation:MinLength=1
+type ImprovedDuration string
```

Comment:
`ImprovedDuration` sounds a bit weird to me 😬, maybe we could think of something that reminds what it is doing?

`LengthValidatedDuration`? `NonEmptyDuration`?

Reply:
On second thoughts I feel it can be different than `ImprovedDuration`

Lets see what @simonpasquier say :)

Reply:
My creativity failed me when I reviewed the code :)
I'm fine with `NonEmptyDuration`.
The other option would be to rename the existing `Duration` type to `LegacyDuration`.

Reply:
I think we can make rename existing Duration in separate change since that would require to update multiple files and tests as well?

Reply:
Agreed for separate PR!